app/questionnaire/router.py

```diff
@@ -87,9 +66,6 @@ def can_access_location(
         Checks whether the location is valid and accessible.
         :return: boolean
         """
-        if location.section_id not in self.enabled_section_ids:
-            return False
```

Comment:
It's not covered in any of the tests (after adding the changes) and doesn't cause any adverse effects after removal in functional tests, it's hard to find this ever used..

Reply:
on main if I replace `return False` with raising an error, this test `test_cant_access_section_disabled` from `test_router` fails - it seems slightly concerning that the changes would stop a line being it if we don't know why, any idea why that test doesn't hit that line anymore with the changes?

Reply:
Ok, it is a question of initialising a clean data store then possibly, since we are within the context of that parent test case? This seems to work (visits the code now):
```python
    def test_cant_access_section_disabled(self):
        self.schema = load_schema_from_name("test_section_enabled_checkbox")
        self.data_stores = DataStores()

        current_location = Location(
            section_id="section-2", block_id="section-2-block", list_item_id=None
        )
        can_access_location = self.router.can_access_location(
            current_location, routing_path=[]
        )

        assert not can_access_location
```

Reply:
ahhh, yeah, makes sense
we can probably do something like this in `RouterTestCase` to make sure data stores is fresh every time
```python
class RouterTestCase:
    schema = None
    data_stores: DataStores

    @pytest.fixture(autouse=True)
    def setup(self):
        self.data_stores = DataStores()

    @cached_property
    def router(self):
        return Router(self.schema, self.data_stores)
```

Reply:
Yeah, makes sense.

Reply:
Yeah I think tests should not depend on each other so good stuff 👍